src/components/Common/ImageUpload/useUploadImages.jsx

```diff
@@ -0,0 +1,199 @@
+    };
+
+    const handleDragOver = (e) => {
+        e.preventDefault();
+    };
+
+    const handleSubmit = async (file) => {
+        try {
+            const formData = new FormData();
+            formData.append("image", file);
+            const API_URL = getApiUrl("api/upload");
+            await axios.post(API_URL, formData);
+        } catch (err) {
+            if (err.message === "Request failed with status code 429") {
+                toast.error("You are uploading images too fast. Please wait a few seconds and try again.");
+            }
+        }
+    };
+
+    const handleSubmitManyImages = async (files) => {
+        for (const file of files) {
+            await handleSubmit(file);
+        }
+    };
+
+    const handleSubmitFromContent = async (content) => {
+        const imageTags = content.match(/<img[^>]*src="([^"]*)"[^>]*>/g) || [];
+        let updatedContent = content;
+        if (imageTags.length > 0) {
+            async function uploadContentImages() {
+                const imageUrls = imageTags.map((tag) => {
+                    const match = tag.match(/src="([^"]*)"/);
+                    return match ? match[1] : null;
+                });
+
+                for (const imageUrl of imageUrls) {
+                    if (!isImageUrl(imageUrl)) {
+                        await uploadContentImage(imageUrl);
+                    }
+                }
+            }
+
+            await uploadContentImages();
+        }
+
+        async function uploadContentImage(imageUrl) {
+            const [imageType, base64Data] = imageUrl.split(";base64,");
+            const filename = `${pageName}-${Date.now()}.${imageType.split("/")[1]}`;
+            updatedContent = updatedContent.replace(imageUrl, filename.split("-")[1]);
+
+            const newFile = bufferToFile(base64Data, filename, imageType);
+
+            await handleSubmit(newFile);
+        }
+        return updatedContent;
+    };
+
+    return {
+        images,
+        setImages,
+        imagesName,
+        setImagesName,
+        onImageRemove: handleRemove,
+        onImageChange: handleChange,
+        onImageDrop: handleDrop,
+        onImagePaste: handlePaste,
+        onImageDragOver: handleDragOver,
+        onImageSubmit: handleSubmit,
+        onManyImageSubmit: handleSubmitManyImages,
+        onImageFromContentSubmit: handleSubmitFromContent,
+    };
+};
+export default useUploadImages;
+
+function isImageUrl(url) {
+    return (
+        url.startsWith("http://") || url.startsWith("https://") || url.replace(/^data:/, "").startsWith("data:image")
+    );
+}
+
+function bufferToFile(base64Data, fileName, imageType) {
+    const byteCharacters = atob(base64Data);
+
+    const byteNumbers = new Array(byteCharacters.length);
+    for (let i = 0; i < byteCharacters.length; i++) {
+        byteNumbers[i] = byteCharacters.charCodeAt(i);
+    }
+    const uint8Array = new Uint8Array(byteNumbers);
```

Comment:
You can replace those 4 rows into a single map:

`byteNumbers = byteCharacters.map((char , index)=> char.charCodeAt(index));`

This is more like a pseudo code, not sure if this exact code will work or it will need changes.